src/controls/EK-mapfishprint.js

```diff
@@ -1,796 +1,764 @@
+            name: layer.get('name'),
+            version: '1',
+            styleProperty: '_gx_style',
+            styles: {
+              0: {
+                fillColor,
+                fillOpacity,
+                strokeColor,
+                strokeWidth,
+                pointRadius,
+                pointFillColor,
+                externalGraphic,
+                graphicOpacity,
+                labelAlign,
+                labelOutlineColor,
+                labelOutlineWidth,
+                fontSize,
+                fontColor,
+                fontBackColor
+              }
             }
+          });
         });
-
-        return legendObjects;
-        
-
-        
+        break;
+      }
+      default:
     }
 
-    // All urls should point to wms service, regardless if layer type is wfs
-    function fetchSourceUrl(layer) {
-        let url;
-        switch (layer.get('type').toUpperCase()) {
-            case "WMS":
-                if (layer.getSource() instanceof TileWMS) {
-                    url = layer.getSource().getUrls()[0];
-                } else if (layer.getSource() instanceof ImageWMS) {
-                    url = layer.getSource().getUrl();
-                }
-                if (url.charAt(0) === "/") {
-                    url = window.location.protocol + "//" + window.location.hostname + url; 
-                }
-                return url;
-                break;
-            case "WFS":
-                let fullUrl = viewer.getSettings().source[layer.get('sourceName')].url;
-                let parsed = fullUrl.split('/');
-                let result = parsed[0] + '//' + parsed[2];
-                return result + '/geoserver/wms';
-                break;
+    return printableLayers;
+  }
+
+  // Builds mapfish-friendly object to from drawn features
+  function drawToPrint(draw) {
+    const styles = {
+      draw: {
+        fillColor: '#0099ff',
+        fillOpacity: 0.1,
+        strokeColor: '#0099ff',
+        strokeWidth: 1.5
+      },
+      point: {
+        externalGraphic: `${viewer.getUrl()}${PinStyle[0].icon.src}`,
+        graphicWidth: 25,
+        graphicHeight: 25
+      }
+    };
+    const featuresWithStyle = draw.features.map((feature) => {
+      const featureWithStyle = { ...feature };
+      if (!featureWithStyle.properties) {
+        featureWithStyle.properties = {};
+      }
+      featureWithStyle.properties.style = 'draw';
+
+      if (featureWithStyle.geometry.type === 'Point') {
+        featureWithStyle.properties.style = 'point';
+      }
+
+      if (featureWithStyle.properties.annonation) {
+        featureWithStyle.properties.style = `draw-${featureWithStyle.properties.annonation}`;
+        styles[`draw-${featureWithStyle.properties.annonation}`] = {
+          fillOpacity: 0.0,
+          strokeOpacity: 0.0,
+          label: featureWithStyle.properties.annonation,
+          fontColor: '#0099ff'
+        };
+      }
+      return featureWithStyle;
+    });
```

Comment:
This was perhaps the most bloody rewrite. Since the original code used a forEach and mutated the feature in the loop which is a big no no in functional programming concept. So We had to create new features in a map instead.